llgo run  _demo/concat,qsort... get error on Linux/Debian

got error
```bash
root@26b213dd93ae:~/llgo/_demo/concat# llgo run .
/usr/bin/ld: /tmp/5142ecf7bbf941cecca2666fa3ba783217fbb0f3a541903a6b3bf31eb8e98a18-d-ca82f9.o: undefined reference to symbol 'pthread_setspecific@@GLIBC_2.17'
/usr/bin/ld: /lib/aarch64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
and in the hello demo，got warning
```bash
root@ef137ede8eb2:~/llgo/_demo/hello# llgo run .
/usr/bin/ld: warning: cannot find entry symbol ad_strip; defaulting to 00000000000006a0
Hello world
```
environment
---
My `GOGCCFLAGS` has the setup -pthread compile option
```bash
GOGCCFLAGS="-fPIC -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build656916144=/tmp/go-build -gno-record-gcc-switches"
```
`dockerfile`
https://github.com/luoliwoshang/DockerFiles/blob/079a220b0021e761677d431851ee3fc41daa89c5/debian/normal/Dockerfile#L1-L34

`go version`
```bash
go version go1.20.14 linux/arm64
```
`clang version`
```
`clang -v`
Ubuntu clang version 17.0.6 (++20231208085846+6009708b4367-1~exp1~20231208085949.74)
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-17/bin
Found candidate GCC installation: /usr/lib/gcc/aarch64-linux-gnu/10
Found candidate GCC installation: /usr/lib/gcc/aarch64-linux-gnu/9
Selected GCC installation: /usr/lib/gcc/aarch64-linux-gnu/10
Candidate multilib: .;@m64
Selected multilib: .;@m64
```
---
When clang is called to perform the link, the -pthread argument is not carried
```bash
# github.com/goplus/llgo/_demo/qsort
clang [-o /usr/local/go/bin/qsort -Wno-override-module -Xlinker -dead_strip /root/.cache/go-build/3e/3ea31842a559c966a6367248b6c56d2e205e0dc7c28b46738d45afe991d759a2-d.ll -lgc /root/llgo/c/bdwgc/llgo_autogen.ll /root/.cache/go-build/f8/f8aed508865c47a93dba4dd7354dbf8e5388571419240892988c8f149663beb3-d.ll /root/.cache/go-build/51/5142ecf7bbf941cecca2666fa3ba783217fbb0f3a541903a6b3bf31eb8e98a18-d.ll]
/usr/bin/ld: /tmp/5142ecf7bbf941cecca2666fa3ba783217fbb0f3a541903a6b3bf31eb8e98a18-d-917fe0.o: undefined reference to symbol 'pthread_setspecific@@GLIBC_2.17'
/usr/bin/ld: /lib/aarch64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
---
The source address for downloading LLVM in the README is for ubuntu 20 (Focal) 
https://github.com/goplus/llgo/blob/3ff5caef9490a885288d85b827acbd1d710227ad/README.md?plain=1#L226-L235

Try to use the source address of the current test environment (debian/bullseye)
https://github.com/luoliwoshang/DockerFiles/blob/3f55e7b88b4edf83d157498f746636ad71950aee/debian/normal/Dockerfile#L1-L39
```sh
echo 'deb http://apt.llvm.org/bullseye/ llvm-toolchain-bullseye-17 main' | sudo tee /etc/apt/sources.list.d/llvm.list
wget -qO - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
```
Unfortunately, the problem was not solved

--- a/internal/build/build.go
+++ b/internal/build/build.go
@@ -292,7 +292,7 @@ func linkMainPkg(pkg *packages.Package, pkgs []*aPackage, runtimeFiles []string,
 	if app == "" {
 		app = filepath.Join(conf.BinPath, name+conf.AppExt)
 	}
-	const N = 5
+	const N = 6
 	args := make([]string, N, len(pkg.Imports)+len(runtimeFiles)+(N+1))
 	args[0] = "-o"
 	args[1] = app
@@ -300,11 +300,13 @@ func linkMainPkg(pkg *packages.Package, pkgs []*aPackage, runtimeFiles []string,
 	args[3] = "-Xlinker"
 	if runtime.GOOS == "darwin" { // ld64.lld (macOS)
 		args[4] = "-dead_strip"
+		args[5] = "" // It's ok to leave it empty, as we can assume libpthread is built-in on macOS.
 	} else { // ld.lld (Unix), lld-link (Windows), wasm-ld (WebAssembly)
 		args[4] = "--gc-sections"
+		args[5] = "-lpthread" // libpthread is built-in since glibc 2.34 (2021-08-01); we need to support earlier versions.
 	}
-	//args[5] = "-fuse-ld=lld" // TODO(xsw): to check lld exists or not
-	//args[6] = "-O2"
+	//args[6] = "-fuse-ld=lld" // TODO(xsw): to check lld exists or not
+	//args[7] = "-O2"
 	needRuntime := false
 	needPyInit := false
 	packages.Visit([]*packages.Package{pkg}, nil, func(p *packages.Package) {

